src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java

```diff
@@ -92,11 +91,16 @@ private static void extractFullIdent(FullIdent full, DetailAST ast) {
                 extractFullIdent(full, firstChild);
             }
             else if (typeOfAst == TokenTypes.DOT) {
-                final DetailAST firstChild = ast.getFirstChild();
-                extractFullIdent(full, firstChild);
-                full.append(".");
-                extractFullIdent(full, firstChild.getNextSibling());
-                appendBrackets(full, ast);
+                extractFullIdentFromDotAst(full, ast);
+            }
+            else if (typeOfAst == TokenTypes.METHOD_CALL) {
+                extractFullIdent(full, ast.getFirstChild());
+            }
+            else if (typeOfAst == TokenTypes.INDEX_OP) {
+                extractFullIdent(full, ast.getFirstChild());
+            }
+            else if (typeOfAst == TokenTypes.TYPECAST) {
+                extractFullIdentFromTypecastAst(full, ast);
```

Comment:
We can but I won't recommend it, the first condition is not a constant.
```java
if (typeOfAst == TokenTypes.LITERAL_NEW              
        && ast.hasChildren()) {                      
    final DetailAST firstChild = ast.getFirstChild();
    extractFullIdent(full, firstChild);              
}                                                    
```
Removing the variable condition i.e. `&& ast.hasChildren()` and rewriting it as-
```java
case TokenTypes.LITERAL_NEW:
    if (ast.hasChildren()) {
        ......
    }
```
would break the current logic as the under the current logic if `TokenTypes.LITERAL_NEW` doesn't have any children, it can still satisfy
`else if (typeOfAst != TokenTypes.ANNOTATIONS)` branch but refactoring it with switch will prohibit it.

If we **have** to refactor it using switch, this should do the job-
```java
switch (typeOfAst) {                                         
    case TokenTypes.DOT:                                     
        extractFullIdentFromDotAst(full, ast);               
        break;                                               
    case TokenTypes.METHOD_CALL:                             
        extractFullIdent(full, ast.getFirstChild());         
        break;                                               
    case TokenTypes.INDEX_OP:                                
        extractFullIdent(full, ast.getFirstChild());         
        break;                                               
    case TokenTypes.TYPECAST:                                
        extractFullIdentFromTypecastAst(full, ast);          
        break;                                               
    default:                                                 
        if (typeOfAst == TokenTypes.LITERAL_NEW              
            && ast.hasChildren()) {                          
            final DetailAST firstChild = ast.getFirstChild();
            extractFullIdent(full, firstChild);              
        }                                                    
        else if (isArrayTypeDeclarationStart) {              
            full.append(ast);                                
            appendBrackets(full, ast);                       
        }                                                    
        else if (typeOfAst != TokenTypes.ANNOTATIONS) {      
            full.append(ast);                                
        }                                                    
        break;                                               
}                                                            
```

Reply:
I such case, the condition should be split into two `if`s:

```java
    private static void extractFullIdent(FullIdent full, DetailAST ast) {
        if (ast != null) {
            final int typeOfAst = ast.getType();
            switch (typeOfAst) {
                case TokenTypes.DOT:
                    extractFullIdentFromDotAst(full, ast);
                    break;
                case TokenTypes.TYPECAST:
                    extractFullIdentFromTypecastAst(full, ast);
                    break;
                case TokenTypes.LITERAL_NEW:
                    if (ast.hasChildren()) {
                        extractFullIdent(full, ast.getFirstChild());
                    }
                    else {
                        full.append(ast);
                    }
                    break;
                case TokenTypes.METHOD_CALL:
                case TokenTypes.INDEX_OP:
                    extractFullIdent(full, ast.getFirstChild());
                    break;
                case TokenTypes.ANNOTATIONS:
                    break;
                default:
                    full.append(ast);
                    if (isArrayTypeDeclarationStart(ast.getNextSibling())) {
                        appendBrackets(full, ast);
                    }
                    break;
            }
        }
    }
```

Reply:
@Vyom-Yadav ping

Reply:
Good suggestion, implemented, done.

Reply:
The default branch is changed and has no `ANNOTATIONS` case because of the original order of branches in `if-else`. pitest was failing when I changes the branch order.